src/firebase/filtering.js

```diff
@@ -0,0 +1,152 @@
+import { db } from "./firebaseConfig";
+import { collection, getDocs, query, where } from "firebase/firestore";
+
+export async function propertyFiltering(preference) {
+  const properties = collection(db, "properties");
+  var count = 0;
+  // key: id; value: number of satisfied requirements
+  var ids = {};
+
```

Comment:
```suggestion
let queryMap = {["price", ">=", "budgetMin"], ["price", "<=", "budgetMax"], ["numOfBedroom", "==", "numBedrooms"] }
```

Reply:
Thank you for your suggestions. However, I think putting all the info in an array and using a forloop would create more edge cases since the if condition is different for different preference. In that case,  I also need to manually encode all the if condition to an array. Although it makes the code concise, it will make it less readable.